openassessment/data.py

```diff
@@ -670,3 +676,178 @@ def collect_ora2_responses(cls, course_id, desired_statuses=None):
+                    cls._download_file_by_key(file_data['key'])
+                    if file_data['type'] == cls.ATTACHMENT
+                    else file_data['content']
+                )
+
+                zip_file.writestr(file_data['file_path'], file_content)
+
+            downloads_csv = cls._create_downloads_csv(submission_files_data)
+            downloads_csv_path = os.path.join(str(course_id), 'downloads.csv')
+
+            zip_file.writestr(
+                downloads_csv_path,
+                downloads_csv.getvalue().encode('utf-8')
+            )
+
+        file.seek(0)
+
+        return True
+
+    @classmethod
+    def collect_ora2_submission_files(cls, course_id):
+        """
+        Collects information about submission files in form list, where each
+        item represents attachment or submission text.
+        """
+
+        all_submission_information = sub_api.get_all_course_submission_information(course_id, 'openassessment')
+
+        files = []
+
+        for student, submission, _ in all_submission_information:
+            answer = submission['answer']
+
+            # collecting submission attachments with metadata
+            for index, file_key in enumerate(answer.get('file_keys', [])):
+                file_name = answer['files_names'][index]
+
+                files.append({
+                    'type': cls.ATTACHMENT,
+                    'course_id': course_id,
+                    'block_id': student['item_id'],
+                    'student_id': student['student_id'],
+                    'key': file_key,
+                    'name': file_name,
+                    'description': answer['files_descriptions'][index],
+                    'size': answer['files_sizes'][index],
+                    'file_path': os.path.join(
+                        str(course_id),
+                        student['item_id'],
+                        student['student_id'],
+                        'attachments',
+                        file_name,
+                    )
+                })
+
+            # collecting submission answer texts
+            for index, part in enumerate(answer.get('parts', [])):
+                content = part['text']
+
+                file_name = 'part_{}.txt'.format(index)
+
+                files.append({
+                    'type': cls.TEXT,
+                    'course_id': course_id,
+                    'block_id': student['item_id'],
+                    'student_id': student['student_id'],
+                    'key': '',
+                    'name': file_name,
+                    'description': 'Submission text.',
+                    'content': content,
+                    'size': len(content),
+                    'file_path': os.path.join(
+                        str(course_id),
+                        student['item_id'],
+                        student['student_id'],
+                        file_name,
+                    )
+                })
+
+        return files
```

Comment:
An optimization would be to make this method an iterator instead of returning a list of files. Just `yield` each file dict.